### ISSUE TITLE
Road image tearing when zooming near large water bodies

### ISSUE THREAD

**Issue by kikuok** [organicmaps/organicmaps#7450]
There is a bug when the scale of roads near huge bodies of water is glitchy, the scale disappears for seconds, the roads are broken. Tested on 2 phones (Asus zb631kl, Samsung Galaxy Note 9 Android 9) on forest roads along Lake Ladoga. Examples: 60.960808, 30.357142 ; 60.346334, 30.887462
The problem is repeated at different render settings. The problem does not depend on openstreetmap data
![photo_2024-02-21_23-26-18](https://github.com/organicmaps/organicmaps/assets/81811653/4d99f217-0c9d-49b7-9653-ada0323b875a)

**Comment by Jean-BaptisteC** [comment]
Not able to reproduced on Pixel 6 - Android 14
Which is Organic Maps version?

**Comment by pastk** [comment]
Reproduced on Android 9 Xiaomi Mi A2 Lite with both GL and Vulkan drivers, the "gaps" appear for a split second while zooming in-out.

Might be related to the following constant
https://github.com/organicmaps/organicmaps/blob/25a0a08bf5f520f39d9d9daee736c99a37ce6a8a/drape_frontend/line_shape.cpp#L389-L394

**Comment by pastk** [comment]
FYI @vng

**Comment by pastk** [comment]
> Not able to reproduced on Pixel 6 - Android 14 Which is Organic Maps version?

Its much less noticeable on faster devices.

> Reproduced on Android 9 Xiaomi Mi A2 Lite with both GL and Vulkan drivers

E.g. on a more modern 6Gb Snapdragon 730 device its possible to see it if one pays attention.

And Pixel 6 is an even faster device.

**Comment by pastk** [comment]
I'm quite sure its not a rendering bug per se, but a slow render (and its slower on detailed zooms using the most detailed geometry version).

[Ladoga lake](https://www.openstreetmap.org/relation/1754729) is a giant complex multipolygon (its slow to be highlighted on osm.org even).

It takes **840Kb (sic!)** in an mwm. ~170k triangles (all 4 geom levels combined).
[Onega lake](https://www.openstreetmap.org/relation/1308279) is even bigger at 920Kb and 195k triangles.

These giant polygons has to be loaded and processed despite that only a small part of them could be visible in the viewport.

What's even worse - those polygons are included (duplicated) into several neighboring map regions (3 in this case).
- see https://github.com/organicmaps/organicmaps/issues/6532

**Comment by vng** [comment]
Yes, we should split them on smaller ones in the Generator. I remember the same issue with lake Michigan (can't find an issue now).
